src/hash-table.c

```diff
@@ -32,8 +32,9 @@ CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #endif
 
 struct _HashTableEntry {
-	HashTableKey key;
-	HashTableValue value;
+//	HashTableKey key;
+//	HashTableValue value;
+	KeyValuePair* value_pair;
```

Comment:
Please put * next to variable name, not type name.

I think you can just name this 'pair'.